Data/ByteString/Streaming/Char8.hs

```diff
@@ -676,30 +652,36 @@ newline :: Word8
 newline = 10
 {-# INLINE newline #-}
 
+-- | Promote a vanilla `String` into a stream.
 string :: String -> ByteString m ()
 string = chunk . B.pack . Prelude.map B.c2w
 {-# INLINE string #-}
 
-
+-- | Returns the number of times its argument appears in the `ByteString`.
 count_ :: Monad m => Char -> ByteString m r -> m Int
 count_ c = R.count_ (c2w c)
 {-# INLINE count_ #-}
 
+-- | Like `count_`, but suitable for use with `Streaming.mapped`.
 count :: Monad m => Char -> ByteString m r -> m (Of Int r)
 count c = R.count (c2w c)
 {-# INLINE count #-}
 
+-- | /O(1)/ Extract the head and tail of a 'ByteString', or its return value if
+-- it is empty. This is the \'natural\' uncons for an effectful byte stream.
 nextChar :: Monad m => ByteString m r -> m (Either r (Char, ByteString m r))
 nextChar b = do
   e <- R.nextByte b
   case e of
     Left r       -> return $! Left r
     Right (w,bs) -> return $! Right (w2c w, bs)
 
+-- | Print a stream of bytes to STDOUT.
 putStr :: MonadIO m => ByteString m r -> m r
 putStr = hPut IO.stdout
 {-# INLINE putStr #-}
 
+-- | Print a stream of bytes to STDOUT, ending with a final @\n@.
```

Comment:
In the `bytestring` library some folks asked for a note explaining that the the final newline is not added atomically, and in a multi-threaded application another write might happen between the tail of the bytestring and the appended newline. This is likely the case here too...